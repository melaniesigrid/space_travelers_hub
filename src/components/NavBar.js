import React from 'react';
import { Navbar, Nav, Container } from 'react-bootstrap';
import { LinkContainer } from 'react-router-bootstrap';
import planet from '../images/planet.png';

function NavBar() {
  return (
    <Navbar activekey={window.location.pathname} expand="md" sticky="top" variant="light" bg="light" className="border-bottom Navbar">
      <Container>
        <LinkContainer to="/" className="LogoContainer" rel="noopener noreferrer">
          <Navbar.Brand className="SpaceTravelers">
            <img src={planet} alt="Planet logo" className="Logo" />
            Space Travelers&apos; Hub
          </Navbar.Brand>
        </LinkContainer>
        <Nav className="ms-auto d-flex">
          <LinkContainer to="/" rel="noopener noreferrer" className="MyLink">
            <Nav.Link>Rockets</Nav.Link>
          </LinkContainer>

          <LinkContainer to="/Missions/" className="MyLink" rel="noopener noreferrer">
            <Nav.Link>Missions</Nav.Link>
          </LinkContainer>

          <LinkContainer to="/MyProfile/" className="MyLink" rel="noopener noreferrer">
            <Nav.Link>
              <span className="Separator">|</span>
              <span>My Profile</span>
            </Nav.Link>
          </LinkContainer>
        </Nav>
      </Container>

    </Navbar>

  );
}

export default NavBar;
