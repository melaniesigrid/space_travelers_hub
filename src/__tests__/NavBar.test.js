import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import NavBar from '../components/NavBar';

it('Header Component Renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <NavBar />
        </Router>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
