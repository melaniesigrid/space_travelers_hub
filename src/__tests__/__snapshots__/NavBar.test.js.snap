// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Renders correctly 1`] = `
<nav
  activekey="/"
  className="border-bottom Navbar navbar navbar-expand-md navbar-light bg-light sticky-top"
>
  <div
    className="container"
  >
    <a
      className="LogoContainer SpaceTravelers active navbar-brand"
      href="/"
      onClick={[Function]}
      rel="noopener noreferrer"
      style={Object {}}
    >
      <img
        alt="Planet logo"
        className="Logo"
        src="planet.png"
      />
      Space Travelers' Hub
    </a>
    <div
      className="ms-auto d-flex navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        className="MyLink nav-link"
        data-rr-ui-event-key="/space_travelers_hub/"
        disabled={false}
        href="/space_travelers_hub/"
        onClick={[Function]}
        rel="noopener noreferrer"
        style={null}
      >
        Rockets
      </a>
      <a
        className="MyLink nav-link"
        data-rr-ui-event-key="/Missions/"
        disabled={false}
        href="/Missions/"
        onClick={[Function]}
        rel="noopener noreferrer"
        style={null}
      >
        Missions
      </a>
      <a
        className="MyLink nav-link"
        data-rr-ui-event-key="/MyProfile/"
        disabled={false}
        href="/MyProfile/"
        onClick={[Function]}
        rel="noopener noreferrer"
        style={null}
      >
        <span
          className="Separator"
        >
          |
        </span>
        <span>
          My Profile
        </span>
      </a>
    </div>
  </div>
</nav>
`;
